src-tauri/src/command.rs

```diff
@@ -1,39 +1,61 @@
-/// command it receives and delegates its corresponding payload to the designated command function.
+/// This command handler is the single point of entry to the business logic in the backend. It will delegate the
+/// command it receives to the designated functions that modify the state (see: "reducers" in the Redux pattern).
 /// NOTE: Testing command handlers is not possible as of yet, see: https://github.com/tauri-apps/tauri/pull/4752
 #[tauri::command]
-pub async fn execute_command(
-    CommandMessage { command, payload }: CommandMessage,
-    _app_state: tauri::State<'_, AppState>,
+pub async fn handle_action(
+    Action { r#type, payload }: Action,
+    app_handle: tauri::AppHandle,
+    app_state: tauri::State<'_, AppState>,
     window: tauri::Window,
-) -> Result<TransferState, String> {
-    info!("received command `{}` with payload `{}`", command, payload);
+) -> Result<(), String> {
+    info!("received action `{:?}` with payload `{:?}`", r#type, payload);
+
+    // TODO: be more redux-idiomatic: do not deserialize to "known" actions, but return the state unchanged if the action is unknown
 
-    match command.as_str() {
-        "[INIT] Get initial state" => {
-            let initial_state = AppState::new(StateStatus::Stable);
-            let transfer_state = TransferState::from(initial_state);
-            window.emit("state-changed", &transfer_state).unwrap();
-            info!(
-                "emitted event `{}` with payload `{:?}`",
-                "state-changed", &transfer_state
-            );
-            Ok(transfer_state)
+    match r#type {
+        ActionType::GetState => {
+            let transfer_state: TransferState = load_state(app_handle).await.unwrap_or_default();
+
+            // TODO: find a better way to populate all fields with values from json file
+            *app_state.status.lock().unwrap() = transfer_state.status;
+            *app_state.active_profile.lock().unwrap() = transfer_state.active_profile;
+            *app_state.locale.lock().unwrap() = transfer_state.locale;
+        }
+        ActionType::Reset => {
+            reset_state(app_state.inner(), Action { r#type, payload }).unwrap();
+            delete_state(app_handle).await.unwrap();
         }
-        _ => Err(format!("Invalid command: {}", command)),
-    }
+        ActionType::CreateNew => {
+            create_did_key(app_state.inner(), Action { r#type, payload }).unwrap();
+            save_state(app_handle, TransferState::from(app_state.inner()))
+                .await
+                .unwrap();
+        }
+        ActionType::SetLocale => {
+            set_locale(app_state.inner(), Action { r#type, payload }).unwrap();
+            save_state(app_handle, TransferState::from(app_state.inner()))
+                .await
+                .unwrap();
+        }
+    };
+
+    let updated_state = TransferState::from(app_state.inner());
+    emit_event(window, updated_state.clone()).unwrap();
```

Comment:
`updated_state` does not need to be cloned, an we can ignore the `emit_event` `Result`.
```suggestion
    emit_event(window, updated_state).ok();
```